internal/addon/authentication/provider.go

```diff
@@ -179,6 +204,23 @@ func (sp *secretsProvider) injectCA(ctx context.Context, targetAuthType map[Targ
 	return nil
 }
 
+func (sp *secretsProvider) getExistingSecret(ctx context.Context, k client.Client, secretName string) (*corev1.Secret, error) {
+	if sp.DefaultNamespace == "" {
+		return nil, kverrors.New("default namespace not defined in secrets provider config")
+	}
+	existingSecret := &corev1.Secret{}
+	key := client.ObjectKey{Name: secretName, Namespace: sp.clusterName}
+	err := k.Get(ctx, key, existingSecret, &client.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		key = client.ObjectKey{Name: secretName, Namespace: sp.DefaultNamespace}
+		err = k.Get(ctx, key, existingSecret, &client.GetOptions{})
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get existing secret: %w", err)
+	}
+	return existingSecret, nil
+}
```

Comment:
This function is really hard to comprehend because of individual things being named badly:
1. Suggest to rename `DefaultNamespace` to `ConfigResourceNamespace` because CRDs used as config in addon are called ConfigResources.
2. Using `sp.clusterName` is simple to grasp but creates confusion when put in the namespace field. Since the SecretProvider constructor is build using a `ManagedClusterAddon` object as input, I suggest to rename `sp.clusterName` to `sp.managedClusterAddonNamespace`.
3. The fallback behavior need some refactor for better readability as:

```go
switch {
// Fallback lookup of user provided secret in the config resource namespace.
case apierrors.IsNotFound(err):
    key = client.ObjectKey{Name: secretName, Namespace: sp.ConfigResourceNamespace}
    err = k.Get(ctx, key, existingSecret, &client.GetOptions{})
    if err != nil {
        return nil, fmt.Errorf("failed to get existing secret: %w", err)
    }
// Err on any other case when lookup of user provided secret in the managed cluster addon namespace
case err != nil:
    return nil, fmt.Errorf("failed to get existing secret: %w", err)
}

return existingSecret, nil
```